app/models/mixins/dialog_option_mixin.rb

```diff
@@ -0,0 +1,33 @@
+module DialogOptionMixin
+  def self.get_dialog_option(key, value, from)
+    # Return value - Support array and non-array types
+    data = value.nil? ? from[key] : value
+    data.kind_of?(Array) ? data.first : data
+  end
+
+  def get_dialog_option(key, value = nil)
+    DialogOptionMixin.get_dialog_option(dialog_key(key), value, dialog_options)
+  end
+
+  def get_dialog_option_decrypted(key, value = nil)
```

Comment:
We have `MiqPassword.try_decrypt` if that makes any of this easier.  I don't know if you depend on the `ArgumentError` anywhere

Reply:
@bdunne 
For the service dialog the field type is appended to the key name e.g.
password::dialog_ldap_password = {v2}......

This is checking if the key has a **password::** prefix.

Reply:
> This is checking if the key has a password:: prefix.

Does the key need or already have that prefix?

Reply:
@bdunne It does, that is done here https://github.com/ManageIQ/manageiq/blob/c532321e256d57c3e427122fd4efda5420a239f8/app/models/dialog_field_text_box.rb#L39